conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEnclaveEnvironment.kt

```diff
@@ -28,103 +31,146 @@ class GramineEnclaveEnvironment(
         require(enclaveMode != EnclaveMode.MOCK) { "Gramine can't run in MOCK mode" }
     }
 
-    private val tcbLevel = 1
+    override fun createReport(
+        targetInfo: ByteCursor<SgxTargetInfo>?,
+        reportData: ByteCursor<SgxReportData>?
+    ): ByteCursor<SgxReport> {
+        return if (enclaveMode == EnclaveMode.SIMULATION) {
+            createSimulationReport(reportData)
+        } else {
+            val report = retrieveReport(
+                targetInfo?.buffer?.getRemainingBytes(avoidCopying = true) ?: byteArrayOf(),
+                reportData?.buffer?.getRemainingBytes(avoidCopying = true) ?: byteArrayOf()
+            )
+            Cursor.slice(SgxReport, ByteBuffer.wrap(report))
+        }
```

Comment:
Ok, great. This is somehow what we noticed since introducing Gramine, simulation mode is quite different from debug and release. Will split in two classes